src/lib/pmd/PmdEngine.ts

```diff
@@ -30,6 +30,40 @@ interface PmdViolation extends Element {
 	};
 }
 
+type HardcodedRuleDetail = {
+	base: {
+		name: string;
+		category: string;
+	};
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	nodeIdentifier: (node: any) => boolean;
+	severity: number;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	msgFormatter: (node: any) => string;
+};
+
+const HARDCODED_RULE_DETAILS: HardcodedRuleDetail[] = [
+	{
+		base: HARDCODED_RULES.FILES_MUST_COMPILE,
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		nodeIdentifier: (node: any): boolean => {
+			return node.name === 'error' && node.attributes.msg.toLowerCase().startsWith('pmdexception: error while parsing');
+		},
+		severity: 1,
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		msgFormatter: (node: any): string => {
+			// We want to take the message property that the node already has, since that's the one that describes what happened.
+			const topLevelMsg = node.attributes.msg;
+			// We also want the exception that caused this exception to throw, since that's the one that usually has information
+			// about lines and columns.
+			const causationStart = node.elements[0].cdata.indexOf('Caused by: ');
```

Comment:
If we happen to get a weird stacktrace that doesn't have a `Caused by`, I suppose the causationStart could remain undefined and we could get errors in the next couple of lines. Do we need a conditional to avoid this?

Reply:
Good point. I'll add a conditional.